File: Arrays/Derived_Structures/Queue/Implementation/Python/priority_queue_array.py
import heapq

class PriorityQueue:
	def __init__(self):
		self.arr = []
	def push(self, x):
		heapq.heappush(self.arr, -x)
	def pop(self):
		if self.arr:
			heapq.heappop(self.arr)
	def top(self):
		return -self.arr[0] if self.arr else None
	def empty(self):
		return not self.arr

if __name__ == "__main__":
	pq = PriorityQueue()
	pq.push(3); pq.push(1); pq.push(2)
	print(pq.top())
	pq.pop()
	print(pq.top())
	print("Empty" if pq.empty() else "Not Empty")

File: Arrays/Derived_Structures/Queue/Implementation/C++/priority_queue_array.cpp
#include <bits/stdc++.h>
using namespace std;

class PriorityQueue {
	priority_queue<int> pq;
public:
	void push(int x) { pq.push(x); }
	void pop() { if (!pq.empty()) pq.pop(); }
	int top() { return pq.empty() ? -1 : pq.top(); }
	bool empty() { return pq.empty(); }
};

int main() {
	PriorityQueue pq;
	pq.push(3); pq.push(1); pq.push(2);
	cout << pq.top() << endl;
	pq.pop();
	cout << pq.top() << endl;
	cout << (pq.empty() ? "Empty" : "Not Empty") << endl;
	return 0;
}
